### ISSUE TITLE
New-NavContainer.ps1 fails with imported Database Backup

### ISSUE THREAD

**Issue by megel** [Microsoft/navcontainerhelper#374]
I use a database backup (including an App and C/Side changes) to create a new container. I use `SetupNavContainer.ps1` with a modification to Import the database backup:
```PowerShell
#>> Download the DB-Backup file ####
    # Use the DB-Backup file
    if ((Test-Path "c:\myfolder\database.bak" -PathType Leaf) -and ("$sqlServerType" -eq "SQLExpress")) {
        Log "Importing c:\myfolder\database.bak to container"
        $additionalParameters += @('--env bakfile="c:\run\my\database.bak"')
        $addUserToDatabase = $true
    }
#<< ####
```
![image](https://user-images.githubusercontent.com/15800069/53967720-06b26d80-40f6-11e9-80d6-6f0f8e23677e.png)

The scripts fails at `New-NavContainer.ps1`:
![image](https://user-images.githubusercontent.com/15800069/53967748-12059900-40f6-11e9-8ba4-5844332dd026.png)
`$enableSymbolLoading` is true (based on the ARM template `GetBCExt`)

The script works, when I change `New-NavContainer.ps1` to:
```PowerShell
    if ($enableSymbolLoading) {
        # Unpublish symbols when running hybrid development
        Invoke-ScriptInNavContainer -containerName $containerName -scriptblock {
            # Unpublish only, when App is present
            if ((Get-NavAppInfo -ServerInstance NAV -Name "Application" -Publisher "Microsoft") -ne $null) {
                Unpublish-NavApp -ServerInstance NAV -Name "Application" -Publisher "Microsoft"
            }
            # Unpublish only, when App is present
            if ((Get-NavAppInfo -ServerInstance NAV -Name "Test" -Publisher "Microsoft") -ne $null) {
                Unpublish-NavApp -ServerInstance NAV -Name "Test" -Publisher "Microsoft" 
            }
        } 
    }
```

**Comment by freddydk** [comment]
Does that work???
Get-NavAppInfo will return $null unless you add -SymbolsOnly

The right way would probably be:

```
Get-NavAppInfo -ServerInstance NAV -Name "Application" -Publisher "Microsoft" -SymbolsOnly | Unpublish-NavApp
```

**Comment by megel** [comment]
Hi hi sure 😂 in a database without these symbols ... Shame on me! I didn't check the opposite way 😳

**Comment by megel** [comment]
I'll check tomorrow

**Comment by megel** [comment]
Hmm - what I don't understand is, without a database.bak, `New-NavContainer.ps1` goes inside of:

```PowerShell
    if ($enableSymbolLoading) {
        # Unpublish symbols when running hybrid development
        Invoke-ScriptInNavContainer -containerName $containerName -scriptblock {
            Unpublish-NavApp -ServerInstance NAV -Name "Application" -Publisher "Microsoft"
            Unpublish-NavApp -ServerInstance NAV -Name "Test" -Publisher "Microsoft"
        }
    }
```

because, `$enableSymbolLoading` is `true` and the Code is passed.

When I use the fresh database backup, it Fails on the same position:
```PowerShell
    if ($enableSymbolLoading) {
        # Unpublish symbols when running hybrid development
        Invoke-ScriptInNavContainer -containerName $containerName -scriptblock {
            Unpublish-NavApp -ServerInstance NAV -Name "Application" -Publisher "Microsoft"
            Unpublish-NavApp -ServerInstance NAV -Name "Test" -Publisher "Microsoft"
        }
    }
```
```PowerShell
PS C:\demo> c:\demo\DownloadDB.ps1
Download Latest Database Backup...
Name                 BlobType  Length          ContentType                    LastModified         AccessTier SnapshotTime         IsDeleted 
----                 --------  ------          -----------                    ------------         ---------- ------------         --------- 
2019-03-07/databa... BlockBlob 311222272       application/octet-stream       2019-03-07 13:58:03Z Unknown                         False     
Recreate Local Container...
Setup new Nav container
Removing container navserver
Using image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
Country de
Version 13.3.27233.0
Locale de-DE
Importing c:\myfolder\database.bak to container
Running mcr.microsoft.com/businesscentral/onprem:de-ltsc2019 (this will take a few minutes)
NavContainerHelper is version 0.5.0.6
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Using image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
Publishing port 8080
Publishing port 443
Publishing port 7046
Publishing port 7047
Publishing port 7048
Publishing port 7049
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Removing navserver from hosts
Removing C:\ProgramData\NavContainerHelper\Extensions\navserver
Creating Nav container navserver
Version: 13.3.27233.0-de
Platform: 13.0.27183.0
Generic Tag: 0.0.9.0
Container OS Version: 10.0.17763.194 (ltsc2019)
Host OS Version: 10.0.17763.316 (ltsc2019)
Using process isolation
Downloading C:\ProgramData\NavContainerHelper\Extensions\navserver\my\license.flf
Creating container navserver from image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
35d400242640665b687541fc19d85d96878a4ce1bbe148c228fef9f030ccbcd0
Waiting for container navserver to be ready
Initializing...
Starting Container
Hostname is navserver
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Using Database .bak file 'c:\run\my\database.bak'
Certificate File Thumbprint 98E7E6509AC6715A9C653149F2429B82F18748D4
Import Certificate to LocalMachine\my
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to true
Starting Service Tier
Using license file 'c:\run\my\license.flf'
Import License
Creating DotNetCore Web Server Instance
Creating http download site
Creating Windows user student
Setting SA Password and enabling SA
Creating Student as SQL User and add to sysadmin
Creating ClickOnce Manifest
Container IP Address: 172.28.112.212
Container Hostname  : navserver
Container Dns Name  : vm12345.westeurope.cloudapp.azure.com
Web Client          : https://vm12345.westeurope.cloudapp.azure.com/NAV/
Dev. Server         : https://vm12345.westeurope.cloudapp.azure.com
Dev. ServerInstance : NAV
ClickOnce Manifest  : http://vm12345.westeurope.cloudapp.azure.com:8080/NAV

Files:
http://vm12345.westeurope.cloudapp.azure.com:8080/al-2.1.79379.vsix

WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 184 seconds
Ready for connections!
Reading CustomSettings.config from navserver
Container output
Initializing...
Starting Container
Hostname is navserver
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Using Database .bak file 'c:\run\my\database.bak'
Certificate File Thumbprint 98E7E6509AC6715A9C653149F2429B82F18748D4
Import Certificate to LocalMachine\my
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to true
Starting Service Tier
Using license file 'c:\run\my\license.flf'
Import License
Creating DotNetCore Web Server Instance
Creating http download site
Creating Windows user student
Setting SA Password and enabling SA
Creating student as SQL User and add to sysadmin
Creating ClickOnce Manifest
Container IP Address: 172.28.112.212
Container Hostname  : navserver
Container Dns Name  : vm12345.westeurope.cloudapp.azure.com
Web Client          : https://vm12345.westeurope.cloudapp.azure.com/NAV/
Dev. Server         : https://vm12345.westeurope.cloudapp.azure.com
Dev. ServerInstance : NAV
ClickOnce Manifest  : http://vm12345.westeurope.cloudapp.azure.com:8080/NAV

Files:
http://vm12345.westeurope.cloudapp.azure.com:8080/al-2.1.79379.vsix

WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 184 seconds
Ready for connections!
No published package matches the provided arguments.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.5.0.6\ContainerHandling\Invoke-ScriptInNavContainer.ps1:30 char:9
+         Invoke-Command -Session $session -ScriptBlock $scriptblock -A ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Unpublish-NAVApp], InvalidOperationException
    + FullyQualifiedErrorId : MicrosoftDynamicsNavServer$NAV/nav-systemapplication,Microsoft.Dynamics.Nav.Apps.Management.Cmdlets.UnpublishNavApp
    + PSComputerName        : 35d400242640665b687541fc19d85d96878a4ce1bbe148c228fef9f030ccbcd0
```


Why this warning is shown:
```
WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.
```

I created the VM last Friday with docker image `mcr.microsoft.com/businesscentral/onprem:de` and the database backup today.

**Comment by freddydk** [comment]
How was this bak file created?

Få Outlook til iOS<https://aka.ms/o0ukef>
________________________________
Fra: Michael Megel <notifications@github.com>
Sendt: Thursday, March 7, 2019 8:49:35 PM
Til: Microsoft/navcontainerhelper
Cc: Freddy Kristiansen; Comment
Emne: Re: [Microsoft/navcontainerhelper] New-NavContainer.ps1 fails with imported Database Backup (#374)


Hmm - what I don't understand is, without a database.bak, New-NavContainer.ps1 goes inside of:

    if ($enableSymbolLoading) {
        # Unpublish symbols when running hybrid development
        Invoke-ScriptInNavContainer -containerName $containerName -scriptblock {
            Unpublish-NavApp -ServerInstance NAV -Name "Application" -Publisher "Microsoft"
            Unpublish-NavApp -ServerInstance NAV -Name "Test" -Publisher "Microsoft"
        }
    }

because, $enableSymbolLoading is true and the Code is passed.

When I use the fresh database backup, it Fails on the same position:

    if ($enableSymbolLoading) {
        # Unpublish symbols when running hybrid development
        Invoke-ScriptInNavContainer -containerName $containerName -scriptblock {
            Unpublish-NavApp -ServerInstance NAV -Name "Application" -Publisher "Microsoft"
            Unpublish-NavApp -ServerInstance NAV -Name "Test" -Publisher "Microsoft"
        }
    }

PS C:\demo> c:\demo\DownloadDB.ps1
Download Latest Database Backup...
Name                 BlobType  Length          ContentType                    LastModified         AccessTier SnapshotTime         IsDeleted
----                 --------  ------          -----------                    ------------         ---------- ------------         ---------
2019-03-07/databa... BlockBlob 311222272       application/octet-stream       2019-03-07 13:58:03Z Unknown                         False
Recreate Local Container...
Setup new Nav container
Removing container navserver
Using image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
Country de
Version 13.3.27233.0
Locale de-DE
Importing c:\myfolder\database.bak to container
Running mcr.microsoft.com/businesscentral/onprem:de-ltsc2019 (this will take a few minutes)
NavContainerHelper is version 0.5.0.6
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Using image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
Publishing port 8080
Publishing port 443
Publishing port 7046
Publishing port 7047
Publishing port 7048
Publishing port 7049
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Removing navserver from hosts
Removing C:\ProgramData\NavContainerHelper\Extensions\navserver
Creating Nav container navserver
Version: 13.3.27233.0-de
Platform: 13.0.27183.0
Generic Tag: 0.0.9.0
Container OS Version: 10.0.17763.194 (ltsc2019)
Host OS Version: 10.0.17763.316 (ltsc2019)
Using process isolation
Downloading C:\ProgramData\NavContainerHelper\Extensions\navserver\my\license.flf
Creating container navserver from image mcr.microsoft.com/businesscentral/onprem:de-ltsc2019
35d400242640665b687541fc19d85d96878a4ce1bbe148c228fef9f030ccbcd0
Waiting for container navserver to be ready
Initializing...
Starting Container
Hostname is navserver
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Using Database .bak file 'c:\run\my\database.bak'
Certificate File Thumbprint 98E7E6509AC6715A9C653149F2429B82F18748D4
Import Certificate to LocalMachine\my
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to true
Starting Service Tier
Using license file 'c:\run\my\license.flf'
Import License
Creating DotNetCore Web Server Instance
Creating http download site
Creating Windows user student
Setting SA Password and enabling SA
Creating Student as SQL User and add to sysadmin
Creating ClickOnce Manifest
Container IP Address: 172.28.112.212
Container Hostname  : navserver
Container Dns Name  : vm12345.westeurope.cloudapp.azure.com
Web Client          : https://vm12345.westeurope.cloudapp.azure.com/NAV/
Dev. Server         : https://vm12345.westeurope.cloudapp.azure.com
Dev. ServerInstance : NAV
ClickOnce Manifest  : http://vm12345.westeurope.cloudapp.azure.com:8080/NAV

Files:
http://vm12345.westeurope.cloudapp.azure.com:8080/al-2.1.79379.vsix

WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 184 seconds
Ready for connections!
Reading CustomSettings.config from navserver
Container output
Initializing...
Starting Container
Hostname is navserver
PublicDnsName is vm12345.westeurope.cloudapp.azure.com
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Using Database .bak file 'c:\run\my\database.bak'
Certificate File Thumbprint 98E7E6509AC6715A9C653149F2429B82F18748D4
Import Certificate to LocalMachine\my
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to true
Starting Service Tier
Using license file 'c:\run\my\license.flf'
Import License
Creating DotNetCore Web Server Instance
Creating http download site
Creating Windows user student
Setting SA Password and enabling SA
Creating student as SQL User and add to sysadmin
Creating ClickOnce Manifest
Container IP Address: 172.28.112.212
Container Hostname  : navserver
Container Dns Name  : vm12345.westeurope.cloudapp.azure.com
Web Client          : https://vm12345.westeurope.cloudapp.azure.com/NAV/
Dev. Server         : https://vm12345.westeurope.cloudapp.azure.com
Dev. ServerInstance : NAV
ClickOnce Manifest  : http://vm12345.westeurope.cloudapp.azure.com:8080/NAV

Files:
http://vm12345.westeurope.cloudapp.azure.com:8080/al-2.1.79379.vsix

WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.

Initialization took 184 seconds
Ready for connections!
No published package matches the provided arguments.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.5.0.6\ContainerHandling\Invoke-ScriptInNavContainer.ps1:30 char:9
+         Invoke-Command -Session $session -ScriptBlock $scriptblock -A ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Unpublish-NAVApp], InvalidOperationException
    + FullyQualifiedErrorId : MicrosoftDynamicsNavServer$NAV/nav-systemapplication,Microsoft.Dynamics.Nav.Apps.Management.Cmdlets.UnpublishNavApp
    + PSComputerName        : 35d400242640665b687541fc19d85d96878a4ce1bbe148c228fef9f030ccbcd0

Why this warning is shown:

WARNING: You are running a container which is 82 days old.
Microsoft recommends that you always run the latest version of our containers.


I created the VM last Friday with docker image mcr.microsoft.com/businesscentral/onprem:de and the database backup today.

**Comment by megel** [comment]
Nothing special:
`Backup-NavContainerDatabases -containerName $containername -bakFolder "$folder"`

**Comment by freddydk** [comment]
So that database was already stripped from app and test - then it will fail.

Få Outlook til iOS<https://aka.ms/o0ukef>
________________________________
Fra: Michael Megel <notifications@github.com>
Sendt: Thursday, March 7, 2019 9:17:21 PM
Til: Microsoft/navcontainerhelper
Cc: Freddy Kristiansen; Comment
Emne: Re: [Microsoft/navcontainerhelper] New-NavContainer.ps1 fails with imported Database Backup (#374)


Nothing special:
Backup-NavContainerDatabases -containerName $containername -bakFolder "$folder"

**Comment by freddydk** [comment]
The fix is in NavCOntainerhelper 0.5.0.8
